src/figuresClasses.ts

```diff
@@ -1,11 +1,95 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0) {
+      throw new Error(`Side 'a' must be greater than 0, but got ${a}`);
+    }
+
+    if (b <= 0) {
+      throw new Error(`Side 'b' must be greater than 0, but got ${b}`);
+    }
+
+    if (c <= 0) {
+      throw new Error(`Side 'c' must be greater than 0, but got ${c}`);
+    }
+
+    if (!this.isValidTriangle(a, b, c)) {
+      throw new Error("This triangle can't exist.");
+    }
+  }
+
+  private isValidTriangle(a: number, b: number, c: number): boolean {
+    return a + b > c && a + c > b && b + c > a;
+  }
+
+  public getArea(): number {
+    const s: number = (this.a + this.b + this.c) / 2;
+    const area: number = Math.sqrt(
+      s * (s - this.a) * (s - this.b) * (s - this.c),
+    );
+
+    return Math.floor(area * 100) / 100;
```

Comment:
Consider returning the area value without truncating it to two decimal places unless there's a specific requirement for it. Truncating can be done outside of this method if needed for display purposes.